anyio/_backends/_curio.py

```diff
@@ -105,9 +106,10 @@ def __init__(self, deadline: float = math.inf, shield: bool = False):
         if self._timeout_task:
             await self._timeout_task.cancel(blocking=False)
 
-        host_task = await curio.current_task()
-        self._tasks.remove(host_task)
-        _task_states[host_task].cancel_scope = self._parent_scope
+        self._tasks.discard(self._host_task)
```

Comment:
Genuine question: is there a reason not to use `.remove()` here, as we did in the asyncio implementation?

Reply:
No, this is left over from the previous iteration.